src/IO/S3/PocoHTTPClient.cpp

```diff
@@ -259,7 +259,11 @@ void PocoHTTPClient::makeRequestInternal(
                 String error_message;
                 Poco::StreamCopier::copyToString(response_body_stream, error_message);
 
-                response->SetClientErrorType(Aws::Client::CoreErrors::NETWORK_CONNECTION);
+                if (Aws::Http::IsRetryableHttpResponseCode(response->GetResponseCode()))
+                    response->SetClientErrorType(Aws::Client::CoreErrors::NETWORK_CONNECTION);
+                else
+                    response->SetClientErrorType(Aws::Client::CoreErrors::USER_CANCELLED);
```

Comment:
Why do you choose `USER_CANCELLED`, but not `RESOURCE_NOT_FOUND`, for example?

Reply:
> Why do you choose `USER_CANCELLED`, but not `RESOURCE_NOT_FOUND`, for example?

Looks like this is what they do in their curl implementation, check CurlHttpClient.cpp:628. In addition, if you look for `SetClientErrorType` these are only two types used - NETWORK_CONNECTION and USER_CANCELLED